src/sentry/web/forms/accounts.py

```diff
@@ -165,7 +165,7 @@ class PasswordlessRegistrationForm(forms.ModelForm):
     username = AllowedEmailField(
         label=_("Email"),
         max_length=128,
-        widget=forms.TextInput(attrs={"placeholder": "you@example.com"}),
+        widget=forms.TextInput(attrs={"placeholder": "you@example.com", "id": "registration-username"}),
```

Comment:
Long time I don't touch Django code :)
I had to read the docs.

## attrs.id 👍 

Setting the [`id`](https://docs.djangoproject.com/en/1.11/ref/forms/api/#django.forms.BoundField.id_for_label) here is fine, but feels brittle. We had to update 2 fields in 2 different classes, and we have to keep in mind what other fields appear in the other form that is rendered on the same page...

## prefix 👎 

An alternative to consider is to use a [`prefix`](https://docs.djangoproject.com/en/1.11/ref/forms/api/#prefixes-for-forms) when rendering the form or at its definition.

The problem with applying a `prefix` is that it would also change the form field names... and that could break registrations, so better not touch for now. We'd need to guarantee no code depends on the name directly instead of using the Django Forms API (e.g. that piece of JavaScript we're updating didn't use the Forms API).

## auto_id 👍 👍 

The other contender is to use [`auto_id`](https://docs.djangoproject.com/en/1.11/ref/forms/api/#django.forms.Form.auto_id) to let Django set all field IDs consistently.

I suggest uniformly changing all ids in this form by setting a custom `auto_id`:

```python
# src/sentry/web/frontend/auth_login.py

    def get_register_form(self, request, initial=None):
        op = request.POST.get("op")
        return RegistrationForm(
            request.POST if op == "register" else None,
            initial=initial,
            # Custom auto_id to avoid ID collision with AuthenticationForm.
            auto_id="id_registration_%s",
        )
```

Then we don't need to set ids individually:

```suggestion
        widget=forms.TextInput(attrs={"placeholder": "you@example.com"}),
```

Reply:
super! Thank you for explaining all the possible options ... I didn't know about it